tests/ui/static/ref-to-mut-static.rs

```diff
@@ -0,0 +1,90 @@
+//@ revisions: e2021 e2024
+
+//@ [e2021] edition:2021
+//@ [e2024] compile-flags: --edition 2024 -Z unstable-options
+
+#![deny(static_mut_refs)]
+
+use std::ptr::{addr_of, addr_of_mut};
+
+static mut FOO: (u32, u32) = (1, 2);
+
+macro_rules! bar {
+    ($x:expr) => {
+        &mut ($x.0)
+        //[e2024]~^ ERROR creating a mutable reference to a mutable static [E0796]
+        //[e2021]~^^ ERROR creating a mutable reference to mutable static is discouraged [static_mut_refs]
+    };
+}
+
+fn main() {
```

Comment:
In this test, let's please add specific tests for `STATIC +=`, `STATIC *=`, `STATIC /=`, `STATIC -=`, `STATIC |=`, `STATIC &=`, `STATIC ^=`, `STATIC <<=`, `STATIC >>=`, `STATIC[x] =`, `STATIC ==`, `== STATIC`, etc.